### PR TITLE
New frontend design flow

### PR DESCRIPTION
**Pull request by jpakey99**
These changes look to combine the first 3 pages into 1 and simplify the user experience.  ImportImages, SelectFileDestination, and FileRoles are now in 1 file.  In order to make this work, changes were made in other areas to reflect the reduced amount of tabs.  This branch also simplifies the FileRoles by allowing the users to drag and drop the image importer into their roles.  Image bubbles now show the reduced size of the file path and not the full thing.  Note: when removing images, all images assigned a role should stay in their assigned role.

### CODE REVIEW

**Review by sk4900 — APPROVED** [review]

**Review by Victoriam7 — DISMISSED** [review]

File: frontend/src/renderer/components/Process/Dropbox.svelte [hunk 1]
```diff
@@ -9,27 +9,30 @@
...
-<main>
-    <div class="sectionStyle {showError && singleItem && isEmpty(items) ? 'errorStyle' : ''}">
-        <section use:dndzone={{items, flipDurationMs, type, dropFromOthersDisabled: singleItem && items.length > 0}}
-            on:consider={handleSort}
-            on:finalize={handleSort}
-        >
-            {#each items as item (item?.id)}
-                <card animate:flip={{ duration: flipDurationMs }} class="{singleItem ? 'verified' : ''}">
+
+<div class="sectionStyle {singleItem && isEmpty(items) ? 'errorStyle' : ''}">
```
Comment (keenanm500) [inline frontend/src/renderer/components/Process/Dropbox.svelte:19]: careful applying css classes that you removed

File: frontend/src/renderer/components/Process/Dropbox.svelte [hunk 2]
```diff
@@ -9,27 +9,30 @@
...
+    <section use:dndzone={{items, flipDurationMs, type, dropFromOthersDisabled: singleItem && items.length > 0}}
+        on:consider={handleSort}
+        on:finalize={handleSort}
+    >
+        {#each items as item (item?.id)}
+            <card animate:flip={{ duration: flipDurationMs }} class="{singleItem ? 'verified' : ''}">
+                {#if minimum}
```
Comment (keenanm500) [inline frontend/src/renderer/components/Process/Dropbox.svelte:26]: what exactly is this 'minimum' flag for?

**Comment by keenanm500** [comment]
looks good to me for the most part. The way the drop-objects change shape, color, and size when dragged is a little interesting, but doesn't need to be part of this change. Remove the 'errorstyle' class and tell me what 'minimum' is for and this should be good to go

File: frontend/src/renderer/components/Process/Dropbox.svelte (hunk 2 above)
Comment (jpakey99) [reply frontend/src/renderer/components/Process/Dropbox.svelte:26]: The minimum flag is the difference in the way the file is displayed in the uploader section vs file roles.  It allows us to have the little image icon along with the trash icon to remove an individual image.

File: frontend/src/renderer/components/Process/Dropbox.svelte [hunk 3]
```diff
@@ -9,27 +9,30 @@
...
-            {#each items as item (item?.id)}
-                <card animate:flip={{ duration: flipDurationMs }} class="{singleItem ? 'verified' : ''}">
+
+<div class="sectionStyle">
+    <section use:dndzone={{items, flipDurationMs, type, dropFromOthersDisabled: singleItem && items.length > 0}}
+        on:consider={handleSort}
+        on:finalize={handleSort}
+    >
+        {#each items as item (item?.id)}
+            <card animate:flip={{ duration: flipDurationMs }} class="{singleItem ? 'verified' : ''}">
```
Comment (keenanm500) [inline frontend/src/renderer/components/Process/Dropbox.svelte:25]: is this verified thing what makes the image draggable green?

Comment (jpakey99) [reply frontend/src/renderer/components/Process/Dropbox.svelte:25]: I believe so

**Comment by keenanm500** [comment]
The dragging is a little glitchy. It looks like issues with changing the style of the draggables when they are dragged (not sure if this is a good design choice), resizing of the flexbox within the image-upload box (maybe they should be the same size). We can talk more tomorrow if you don't want to deal with this rn

**Review by keenanm500 — APPROVED** [review]

**Pull request closed without merging**